llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -1915,6 +1915,8 @@ ParseStatus RISCVAsmParser::parseCSRSystemRegister(OperandVector &Operands) {
         // Accept an immediate representing a named Sys Reg if it satisfies the
         // the required features.
         for (auto &Reg : Range) {
+          if (Reg.IsAltName || Reg.IsDeprecatedName)
```

Comment:
This isn't strictly needed, but ensures we use the required features from the primary name and not the alternate or deprecated names.